### ISSUE TITLE
meshless_galerkin: backend="jax" is not JAX-traceable (np.asarray at the return breaks vmap and grad)

### ISSUE THREAD

**Issue by derrring** [derrring/MFGArchon#1863]
`shape_functions_and_grads(..., backend="jax")` returns `np.asarray(phi), np.asarray(grad)`
(`mfgarchon/alg/numerical/meshless_galerkin/mls_basis.py:171`). That materializes the result, so the
function is not JAX-traceable: it raises `TracerArrayConversionError` under `jax.vmap` and under
`jax.grad`.

Reproduction:

```python
import numpy as np, jax, jax.numpy as jnp
jax.config.update("jax_enable_x64", True)
from mfgarchon.alg.numerical.meshless_galerkin import mls_basis as lib

nodes = np.linspace(0.0, 1.0, 21).reshape(-1, 1)
exps = lib.monomial_exponents(1, 2)
qp = jnp.asarray(np.linspace(0.3, 0.7, 5).reshape(-1, 1))

jax.vmap(lambda x: lib.shape_functions_and_grads_jax(x[None, :], nodes, 2.6 / 21, exps)[0])(qp)
# jax.errors.TracerArrayConversionError:
#   The numpy.ndarray conversion method __array__() was called on traced array with shape float64[1,21]
```

`jax.grad` on the same closure fails identically. The body itself is fully traceable; only the two
conversions at the return are not.

Why it matters beyond tidiness: a caller that wants the numpy result can call `backend="numpy"`, so
the only reason to select `backend="jax"` is to differentiate or batch *through* the shape
functions — which is exactly what the conversion forbids. The backend currently answers no question
that the default backend does not already answer.

Concretely it blocks a deduplication. `meshless_galerkin/mls_basis.py` descends from a
weak-form-Galerkin prototype that still carries its own copy of the same algorithm, because every
consumer there wraps the shape functions in `jax.vmap` or `jax.grad` and therefore cannot call this
one. The two implementations agree to 4e-13 in phi and 5e-13 in grad today; nothing keeps them
equal.

Suggested fix: return the traced arrays from `shape_functions_and_grads_jax` and let the dispatcher
convert, so `backend="numpy"` keeps its current contract while `backend="jax"` becomes usable under
transformation. If the concrete return type is load-bearing for an existing caller, an explicitly
traceable entry point beside it would do as well.

Note for whoever takes this: the two backends are not interchangeable at rounding level. On a
21-node 1D cloud with degree-2 global monomials and rho = 2.6h, the analytic-derivative numpy path
and the autodiff path separate at 1.6e-9 in the gradient (2e-12 in 2D). That is the conditioning of
the unshifted moment matrix, not a defect, but a test that swaps backends and asserts equality at
1e-12 will fail.
